R/Filter.R

```diff
@@ -619,6 +619,254 @@ makeFilter(
   }
 )
 
+# Preprocess the data before applying praznik filter. According to Praznik documentation:
+# The best practice is to convert data to factors prior to feeding them in this function.
+# Real attributes are cut into about 10 equally-spaced bins, following the heuristic often used in literature.
+# Precise number of cuts depends on the number of objects; namely, it is n/3, but never less than 2 and never more than 10.
+preprocess.cmi.praznik = function(data) {
+    colns = colnames(data)
+    data = convertDataFrameCols(data, logicals.as.factor = TRUE, chars.as.factor = TRUE)
+    int.yes = vlapply(data, is.integer)
+    if (any(int.yes)) data[int.yes] = lapply(data[int.yes], as.factor)
+    numeric.yes = vlapply(data, is.numeric)
+    df.num = data.frame(data[, numeric.yes])
+    interval = max(min(as.integer(nrow(data) / 3L), 10L), 2L)
+    const.cols.yes = vlapply(df.num, function(x) length(unique(x)) < 2L)
+    if (any(const.cols.yes)) {
+      df.num.uni =  lapply(df.num[const.cols.yes], as.factor)
+      df.num.cut = df.num[!const.cols.yes]
+      df.num.cut = data.frame(apply(df.num.cut, 2L, cut, interval))
+      df.num = cbind(df.num.cut, df.num.uni)
+    }
+    colnames(df.num) = colns[numeric.yes]
+    df.nonnum = data.frame(data[, !numeric.yes])
+    colnames(df.nonnum) =  colns[!numeric.yes]
```

Comment:
Where do we loose the colnames? Do we have to re-set them?

Reply:
I just reset them to make them consistent with original data

Reply:
we lose the colnames in line 643